react/modules/tasks/functional-routing.md

```diff
@@ -0,0 +1,80 @@
+# React: Routing and Hooks
+
+## 🧠 Task Description
+
+In this task, you will refactor your class components to functional components with hooks and implement routing functionality. You'll add pagination, detailed views, and URL-based navigation.
+
+## 🎯 Task Goals
+
+- Convert class components to functional components using React hooks
+- Implement React Router for navigation and URL management
+- Create custom hooks for shared logic
+- Add pagination with URL synchronization
+
+## Application Requirements
+
+1. **Pagination**:
+
+   - Add pagination for your item list (search results)
+   - Display current page in URL using query parameters (e.g., `?page=2`)
+   - Pagination should appear after receiving the list of items
+
+2. **Master-Detail View**:
+
+   - Main page displays search results
+   - On item click, split page into 2 sections:
+     - **Left**: Continue displaying search results
+     - **Right**: Display item details using React Router outlet
+   - Show loading indicator while fetching details
+   - Add close control for details section
+   - Reflect details state in URL (e.g., `/?page=2&details=1`)
+
+3. **About Page**:
+   - Create an About page with application author information
+   - Include a link to the RS School React course
+   - Add navigation to reach the About page from the main application
+
+## Technical Requirements
+
+1. **Branch Management**: Create branch from **"unit-testing"**. Branch name: **"hooks-and-routing"**
+
+2. **Component Conversion**:
+
+   - Convert ALL class components to **functional components** with hooks
+   - **Exception**: Keep **Error Boundary** as class component
+   - Create custom hook for localStorage operations
+
+3. **Routing Setup**:
+   - Add **React Router** in SPA mode
```

Comment:
In React Router now there are 3 modes, where 2 are for the SPA: data and declarative.
https://reactrouter.com/start/modes
To avoid confusion among students and to minimize the possible hassle I sugegst phrasing this as:
```
  - Add **React Router** in either "Data" or "Declarative" modes (pick one)
```